apps/system/test/unit/child_window_factory_test.js

```diff
@@ -404,6 +406,48 @@ suite('system/ChildWindowFactory', function() {
     sinon.assert.calledWith(app1._setVisibleForScreenReader, true);
   });
 
+  suite('Hidden popups', function() {
+    test('app with no permissions cannot open hidden popup', function() {
+      var app1 = new MockAppWindow(fakeAppConfig1);
+      var spy = this.sinon.spy(window, 'AppWindow');
+      this.sinon.stub(app1, 'hasPermission').returns(false);
+      var cwf = new ChildWindowFactory(app1);
+      var stubCreatePopupWindow = this.sinon.stub(cwf, 'createPopupWindow');
+      cwf.handleEvent(new CustomEvent('mozbrowseropenwindow',
+        {
+          detail: fakeWindowOpenHiddenPopup
+        }));
+      assert.equal(stubCreatePopupWindow.callCount, 0);
+      assert.equal(spy.callCount, 0);
+    });
+
+    test('app with permissions can open hidden popup', function() {
+      var app1 = new MockAppWindow(fakeAppConfig1);
+      this.sinon.stub(app1, 'hasPermission').returns(true);
+      var cwf = new ChildWindowFactory(app1);
+      var stubCreatePopupWindow = this.sinon.stub(cwf, 'createPopupWindow');
+      cwf.handleEvent(new CustomEvent('mozbrowseropenwindow',
+        {
+          detail: fakeWindowOpenHiddenPopup
+        }));
+      assert.isTrue(stubCreatePopupWindow.calledOnce);
+    });
+
+    test('must setup stayBackground', function() {
+      var app1 = new MockAppWindow(fakeAppConfig1);
+      this.sinon.stub(app1, 'hasPermission').returns(true);
+      var spyPopupWindow = this.sinon.spy(window, 'PopupWindow');
+      var cwf = new ChildWindowFactory(app1);
+      var spyCreatePopupWindow = this.sinon.spy(cwf, 'createPopupWindow');
+      cwf.handleEvent(new CustomEvent('mozbrowseropenwindow',
+        {
+          detail: fakeWindowOpenHiddenPopup
+        }));
+      assert.isTrue(spyCreatePopupWindow.calledOnce);
+      assert.isTrue(spyPopupWindow.getCall(0).args[0].stayBackground);
+    });
```

Comment:
and then extract those tests to run them once against `alwaysLowered` and once against `z-lock` :)
(or we can remove the `z-lock` support from the code, I'm fine with that too)